src/cyclic_execution.rs

```diff
@@ -5,25 +5,48 @@
 use crate::aerugo::Aerugo;
 use crate::data_provider::DataProvider;
 use crate::tasklet::TaskletPtr;
-use crate::time::MillisDurationU32;
+use crate::time::{Duration, Instant};
+use crate::Mutex;
 
 /// Cyclic execution information.
 pub(crate) struct CyclicExecution {
+    /// Next execution time.
+    next_execution_time: Mutex<Instant>,
+    /// Period of cyclic execution.
+    period: Option<Duration>,
     /// Tasklet subscribed for cyclic execution.
     tasklet: TaskletPtr,
-    /// Period of cyclic execution.
-    period: Option<MillisDurationU32>,
 }
 
 impl CyclicExecution {
     /// Creates new instance.
-    pub(crate) fn new(tasklet: TaskletPtr, period: Option<MillisDurationU32>) -> Self {
-        CyclicExecution { tasklet, period }
+    pub(crate) fn new(tasklet: TaskletPtr, period: Option<Duration>) -> Self {
+        CyclicExecution {
+            next_execution_time: Instant::from_ticks(0).into(),
+            period,
+            tasklet,
+        }
     }
 
-    /// Wakes that stored tasklet.
-    pub(crate) fn wake_tasklet(&self) {
-        Aerugo::wake_tasklet(&self.tasklet);
+    /// Wakes that stored tasklet if the time has come.
```

Comment:
```suggestion
    /// Wakes that stored tasklet if the time for it's execution has come.
```

"if the time has come" sounds pretty ominous